Brokerages/Bitfinex/BitfinexSubscriptionManager.cs

```diff
@@ -94,64 +123,92 @@ public void Subscribe(Symbol symbol)
         /// <summary>
         /// Removes the subscription for the requested symbol
         /// </summary>
-        /// <param name="symbol">The symbol</param>
-        public void Unsubscribe(Symbol symbol)
+        /// <param name="symbols">symbol list</param>
+        /// <param name="tickType">Type of tick data</param>
+        protected override bool Unsubscribe(IEnumerable<Symbol> symbols, TickType tickType)
         {
-            List<BitfinexWebSocketWrapper> subscriptions;
-            if (_subscriptionsBySymbol.TryGetValue(symbol, out subscriptions))
+            string channelName = ChannelNameFromTickType(tickType);
+            var states = new List<bool>(symbols.Count());
+            foreach (var symbol in symbols)
             {
-                foreach (var webSocket in subscriptions)
+                List<BitfinexWebSocketWrapper> subscriptions;
+                if (_subscriptionsBySymbol.TryGetValue(symbol, out subscriptions))
                 {
-                    try
+                    for (int i = subscriptions.Count - 1; i >= 0; i--)
                     {
-                        lock (_locker)
+                        var webSocket = subscriptions[i];
+                        try
                         {
-                            List<BitfinexChannel> channels;
-                            if (_channelsByWebSocket.TryGetValue(webSocket, out channels))
+                            lock (_locker)
                             {
-                                foreach (var channel in channels)
+                                Channel channel = new Channel(channelName, symbol);
+                                List<Channel> channels;
+                                if (_channelsByWebSocket.TryGetValue(webSocket, out channels) && channels.Contains(channel))
                                 {
-                                    UnsubscribeChannel(webSocket, channel.ChannelId);
+                                    UnsubscribeChannel(webSocket, channel);
+
+                                    if (_onUnsubscribeEvent.WaitOne(TimeSpan.FromSeconds(30)))
```

Comment:
Believe `_onUnsubscribeEvent.WaitOne` will always timeout, we are taking the `_locker` here and the path setting `_onUnsubscribeEvent` needs the lock too. Is there a reason to need this `_onUnsubscribeEvent`? If not I'd suggest removing it, the simpler the merrier